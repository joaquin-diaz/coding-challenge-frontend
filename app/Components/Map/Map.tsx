import * as React from 'react';
import styled from '@emotion/styled';
import { Props as GoogleMapProps, default as GoogleMapReact } from 'google-map-react';

import Coordinate from '../../types/Coordinate';

const MapContainer = styled.div`
  height: 100vh;
  width: 100%;
`;

interface MapProps {
  center: Coordinate;
}

const Map: React.FC<MapProps> = props => {
  const mapProps: GoogleMapProps = {
    center: props.center,
    zoom: 14,
    bootstrapURLKeys: {
      key: ENV.GOOGLE_MAPS_API_KEY,
    },
  };

  return (
    <MapContainer>
      <GoogleMapReact {...mapProps}>{props.children}</GoogleMapReact>
    </MapContainer>
  );
};

export default Map;
