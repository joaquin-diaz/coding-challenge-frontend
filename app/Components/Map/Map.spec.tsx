import * as React from 'react';

import Map from './Map';
import { shallow } from 'enzyme';

describe('<Map />', () => {
  it('should render', () => {
    const wrapper = shallow(<Map center={{ lat: 123, lng: 456 }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
