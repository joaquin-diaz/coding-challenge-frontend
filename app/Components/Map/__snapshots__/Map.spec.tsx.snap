// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Map /> should render 1`] = `
<Styled(div)>
  <GoogleMap
    bootstrapURLKeys={
      Object {
        "key": "Test Key",
      }
    }
    center={
      Object {
        "lat": 123,
        "lng": 456,
      }
    }
    debounced={true}
    distanceToMouse={[Function]}
    googleMapLoader={[Function]}
    heatmap={Object {}}
    heatmapLibrary={false}
    hoverDistance={30}
    layerTypes={Array []}
    options={[Function]}
    style={
      Object {
        "height": "100%",
        "margin": 0,
        "padding": 0,
        "position": "relative",
        "width": "100%",
      }
    }
    yesIWantToUseGoogleMapApiInternals={false}
    zoom={14}
  />
</Styled(div)>
`;
